modules/minigl/src/test/java/org/jpos/gl/TestBase.java

```diff
@@ -18,17 +18,20 @@
 
 package org.jpos.gl;
 
-import junit.framework.TestCase;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
 
-public abstract class TestBase extends TestCase {
+public abstract class TestBase {
     protected GLSession gls;
     protected long start;
     protected long checkpoint;
 
+    @BeforeEach
```

Comment:
This should get executed first, see [here](https://junit.org/junit5/docs/current/api/org/junit/jupiter/api/BeforeEach.html) for details under `Inheritance and Execution Order`.

Reply:
Hi James, it does not work as you say, as you've overriden the setUp method. For it to work the way to want it, you'd need to change the name of the method in TestBase, so Jupiter executes the @BeforeEach method on the base class, and then the @BeforeEach on the test class.

Reply:
Pushed an update renaming those methods in `TestBase`.

Reply:
While it's OK to rename the method, I believe it's a confusing practice to use annotations like this in the base classes. This is a small class hierarchy, so no problem, but in a bigger one, it becomes confusing to research what the heck is being called by JUnit. The `super.xxx` construct is well known, and easy to follow.

Reply:
From what I could tell it's recommended not to use `super.xxx` in `junit5`.

Reply:
Hi @jameshilliard can you point a reference to this fact? So we know why are the reasons if that is the case?

Reply:
From what I could tell using `super.xxx` explicitly was necessary due to the lack of annotation such as `@BeforeEach` being available for use directly in the superclass. There is also [this](https://github.com/junit-team/junit5/issues/960) which changed some of the annotation inheritance behavior.

Reply:
> While it's OK to rename the method, I believe it's a confusing practice to use annotations like this in the base classes. This is a small class hierarchy, so no problem, but in a bigger one, it becomes confusing to research what the heck is being called by JUnit. The `super.xxx` construct is well known, and easy to follow.

I see your point, yet I'd like to express that using the annotation in the base class has many benefits:

* It can be parsed by your favorite IDE, or tool, making your code easier to follow.
* It makes the execution consistent, in other words it will always execute before/after each test. 
* In the same line, it's impossible to forget to call super.xxxx which could potentially introduce errors in your code.

Reply:
>In the same line, it's impossible to forget to call super.xxxx which could potentially introduce errors in your code.

But I find more error prone the possibility to write a method named the same as the extended class (we always tend to use same names) and forget that we have to change it in order to not affect the super implementation calling.

I prefer the annotations visible closer to where the action happens, in this case renaming the method to not collide with the super class sounds too artificial to me.

Reply:
I have to agree with Andres here, IDE also can deal with calls to super, and copy & paste can land you using the same name, as we just saw.